test/e2e/apimachinery/flowcontrol.go

```diff
@@ -317,6 +335,58 @@ func createFlowSchema(f *framework.Framework, flowSchemaName string, matchingPre
 	}
 }
 
+func getFlowSchema(f *framework.Framework, name string) *flowcontrol.FlowSchema {
+	fs, err := f.ClientSet.FlowcontrolV1beta1().FlowSchemas().Get(context.TODO(), name, metav1.GetOptions{})
+	framework.ExpectNoError(err)
+	return fs
+}
+
+func getPriorityLevelConfig(f *framework.Framework, name string) *flowcontrol.PriorityLevelConfiguration {
+	pl, err := f.ClientSet.FlowcontrolV1beta1().PriorityLevelConfigurations().Get(context.TODO(), name, metav1.GetOptions{})
+	framework.ExpectNoError(err)
+	return pl
+}
+
+// waitForSteadyState repeatedly polls the API server to check if the newly
+// created flow schema and priority level have been seen by the APF controller
+// by checking: (1) the dangling priority level reference condition in the flow
+// schema status, and (2) metrics. The function times out after 30 seconds.
+func waitForSteadyState(f *framework.Framework, flowSchemaName string, priorityLevelName string) {
+	framework.ExpectNoError(wait.Poll(time.Second, 30*time.Second, func() (bool, error) {
+		fs := getFlowSchema(f, flowSchemaName)
+		if apihelpers.GetFlowSchemaConditionByType(fs, flowcontrol.FlowSchemaConditionDangling) != nil {
+			return false, nil
+		}
+		resp, err := f.ClientSet.CoreV1().RESTClient().Get().RequestURI("/metrics").DoRaw(context.TODO())
+		if err != nil {
+			return false, err
+		}
+		sampleDecoder := expfmt.SampleDecoder{
```

Comment:
`getPriorityLevelConcurrency` function already gets the concurrency for a priority level, can we reuse it?

Reply:
The problem with that func, in the current version of this PR, is that when the sought time series is _not_ found the test fails rather than waits and tries again.

Reply:
refactored things around slightly to return an error instead so that we can re-use this